homeassistant/components/vilfo/sensor.py

```diff
@@ -0,0 +1,105 @@
+"""Support for Vilfo Router sensors."""
+from homeassistant.helpers.entity import Entity
+
+from .const import (
+    ATTR_ICON,
+    ATTR_LABEL,
+    ATTR_ROUTER_MANUFACTURER,
+    ATTR_ROUTER_MODEL,
+    ATTR_ROUTER_NAME,
+    ATTR_UNIT,
+    DOMAIN,
+    SENSOR_TYPES,
+)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Add Vilfo Router entities from a config_entry."""
+    vilfo = hass.data[DOMAIN][config_entry.entry_id]
+
+    sensors = []
+
+    name = ATTR_ROUTER_NAME
+    device_info = {
+        "identifiers": {(DOMAIN, vilfo.host, vilfo.mac_address)},
+        "name": name,
+        "manufacturer": ATTR_ROUTER_MANUFACTURER,
+        "model": ATTR_ROUTER_MODEL,
+        "sw_version": vilfo.firmware,
+    }
+
+    sensors.append(VilfoRouterLoadSensor(vilfo, device_info))
+    sensors.append(VilfoRouterUptimeSensor(vilfo, device_info))
```

Comment:
IF you want:

- You can remove your non generic sensor classes
- Add a `type` param to  `VilfoRouterSensor`
- Foreach `SENSOR_TYPES` and pass the key to the `type param.`

Reply:
Great suggestion! I did that along with some other refactoring which made it much more clear and to the point 👍 😄